crates/dojo/lang/src/cairo_plugin.rs

```diff
@@ -25,7 +25,10 @@ pub struct BuiltinDojoPlugin;
 pub fn dojo_plugin_suite() -> PluginSuite {
     let mut suite = PluginSuite::default();
 
-    suite.add_plugin::<BuiltinDojoPlugin>().add_inline_macro_plugin::<SelectorFromTagMacro>();
+    suite
+        .add_plugin::<BuiltinDojoPlugin>()
+        .add_inline_macro_plugin::<SelectorFromTagMacro>()
+        .add_inline_macro_plugin::<BytearrayHashMacro>();
```

Comment:
@bengineer42 fyi the issue was that you were trying to add 2 times the `BuiltinDojoPlugin`. This was causing the weird scarb issue on events, which is not related at all. But this was the reason.